File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/ModelApi.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class ModelApi {
    private final HttpClient client;
    
    public ModelApi(HttpClient client) {
        this.client = client;
    }

    /** Retrieve model */
    public OpenAiModel retrieve(String model) throws Exception {
        return (OpenAiModel) client.get(ApiPaths.aiPath("/models/" + model + ""));
    }

    /** Delete model */
    public OpenAiModelDeleteResponse deleteModel(String model) throws Exception {
        return (OpenAiModelDeleteResponse) client.delete(ApiPaths.aiPath("/models/" + model + ""));
    }

    /** List models */
    public OpenAiModelList listModels() throws Exception {
        return (OpenAiModelList) client.get(ApiPaths.aiPath("/models"));
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/ModelApi.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class ModelApi(private val client: HttpClient) {

    /** Retrieve model */
    suspend fun retrieve(model: String): OpenAiModel? {
        return client.get(ApiPaths.aiPath("/models/$model")) as? OpenAiModel
    }

    /** Delete model */
    suspend fun deleteModel(model: String): OpenAiModelDeleteResponse? {
        return client.delete(ApiPaths.aiPath("/models/$model")) as? OpenAiModelDeleteResponse
    }

    /** List models */
    suspend fun listModels(): OpenAiModelList? {
        return client.get(ApiPaths.aiPath("/models")) as? OpenAiModelList
    }
}
